src/cfg/mod.rs

```diff
@@ -0,0 +1,98 @@
+use docopt::Docopt;
+use slog::Logger;
+use std::fs::File;
+use std::io::Read;
+use toml::from_str;
+
+pub mod errors {
+    error_chain!{}
+}
+
+use self::errors::*;
+
+const CONFIGFILE: &'static str = "blacklung.cfg";
+
+const USAGE: &'static str = "
+Blacklung server.
+
+Usage:
+    blacklung [--port=<port>]
+    blacklung (-h | --help)
+
+Options:
+    -h --help       Show this screen.
+    --port=<PORT>   Port to bind to. Needs to be detailed here or in configuration file.
+";
+
+/// This struct is used for the argument parsing.
+#[derive(Debug, RustcDecodable)]
+struct Args {
+    flag_port: Option<u16>,
+}
+
+/// This struct holds data that can be read from the config file;
+/// hence the Option.
+#[derive(Debug, Deserialize)]
+struct ConfigurationFile {
+    pub port: Option<u16>,
+}
```

Comment:
type PortType = Option<u16>;

and then use PortType in the Args and ConfigurationFile structs.